### DISCUSSION TITLE
How to add hover effects to Google built-in POIs?

### DISCUSSION THREAD

**Discussion by taiga-forestry** [visgl/react-google-maps#653] [Q&A] [answered]
Using maps.google.com, if I hover over a POI (point of interest), the cursor changes to indicate a clickable item, the text changes color, and a pop-up is shown. For instance, consider the case of me hovering over "BCD Tofu House":

<img width="600" alt="Screenshot 2024-12-30 at 8 06 24 PM" src="https://github.com/user-attachments/assets/1cb1c47e-52e9-482c-8811-91a320fd3b99" />

However, when a POI is hovered over in the react-google-maps version, the cursor changes, but the text color does not (and no pop-up is shown).

<img width="600" alt="Screenshot 2024-12-30 at 8 13 08 PM" src="https://github.com/user-attachments/assets/acd3decb-b1f1-4bd6-a99b-cd469a2dae2b" />

This seems like it _should_ be a relatively simple thing to do, but I can't find anything in documentation or online examples on how this can be done. For instance, the _onMouseover_ events in the Map API only trigger when the Map is moused over, not individual POIs, so it doesn't help at all.

I've managed to implement a custom pop-up on **click** using the onClick property, like this:

```
        onClick={(e) => {
          // disable default pop-up when clicking Google markers
          e.stop();

          const {
            detail: { placeId },
          } = e;

          if (placeId) {
            // use placesService to get details like name, address, and trigger state updates
            getPlaceByPlaceID(placesService, placeId, onPlaceSelect);
          }
        }} 
```

but there doesn't seem to be an equivalent way of doing so with mouse hovering/movement.

Honestly, just changing the text color of the POI on hover (like Google Maps browser version already does) without the pop-up on hover would be enough for my use case to help indicate clickability. Any help or ideas with this would be greatly appreciated!

**Comment by usefulthink** [comment, marked as answer]
Unfortunately, that is something that can't currently be done with the Maps JavaScript API. Only the [`click` event](https://developers.google.com/maps/documentation/javascript/reference/map#Map.click) uses the `IconMouseEvent` that contains the placeId when a place-icon is clicked. Besides that, I don't think there is any way to even know if there are place-icons on the map.

A feature request like this would have to be implemented in the Maps API itself: https://issuetracker.google.com/issues?q=status:open%20componentid:188853&s=created_time:desc

**Comment by TheSanDiegonite** [comment]
Also looking for this functionality.  Pretty shocked it doesn't exist.
